AppServices/Services/CustomerAppService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using AppServices.Interfaces;
+using DomainModels;
+using DomainServices.Interfaces;
+using System;
+using System.Collections.Generic;
+
+namespace AppServices.Services
+{
+    public class CustomerAppService : ICustomerAppService
+    {
+        private readonly ICustomerService _service;
+        public CustomerAppService(ICustomerService service)
+        {
+            _service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        public virtual void Create(Customer customer)
+        {
```

Comment:
Mesmo caso da controller, somente usaríamos o virtual se essa classe fosse herdada em outra, como não é o caso, o uso é dispensável.
```suggestion
        public void Create(Customer customer)
        {
```